lib/ansible/modules/identity/keycloak/keycloak_realm.py

```diff
@@ -0,0 +1,1059 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, Eike Frost <ei@kefro.st>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: keycloak_realm
+
+short_description: Allows administration of Keycloak realms via Keycloak API
+
+version_added: "2.7"
```

Comment:
I think this should be 2.8 by now. I don't think new features are released with bugfix updates. See https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html

Reply:
It was current when I submitted the PR and it passed the review process. Since then nothing has happened. I'll gladly update the version number (will be at a workstation for it tomorrow), though.

Reply:
As to integration tests, there seemed to be some pushback to the idea of doing that within the ansible testing infrastructure on the IRC development channel, citing the added time and maintenance a full test run would take -- so I didn't pursue it any further.

From what I saw there were a few minor updates to the API with newer Keycloak versions. I'm happy to incorporate them as well (though the API documentation for Keycloak is quite horrendous), though my initial plan was to do that as maintenance once this patch was in, as I didn't want to trigger another full review wasting time for all involved for, as you said, minor changes.

Reply:
My own integration test for basic keycloak role behavior takes around 15-30 minutes which is very heavy. I think it is valid to say that you do not want those test it the infrastructure does not support it very well.
Furthermore this ansible modules should either support Keycloak 3 or 4. If neccesary we can extend on the existing modules and create a keycloak4_realm module from that if e.g. arguments change.

Also thought again about testing. As of now the Keycloak API and modules are in an unstable phase and subject to change. Writing tests for interfaces that may constantly change is very tiresome. As of now i see that the keycloak.py in the modul_utils begins to bloat with duplicated code (get_realm, create_realm, get_client, create_client, ....). Maybe it is better to improve on that later on.

So lets shipit. Thank you very much for your effort :)